gazebo_ros/CMakeLists.txt

```diff
@@ -71,13 +71,13 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GAZEBO_CXX_FLAGS}")
 
 ## Plugins
 add_library(gazebo_ros_api_plugin src/gazebo_ros_api_plugin.cpp)
-add_dependencies(gazebo_ros_api_plugin gazebo_msgs_gencpp ${PROJECT_NAME}_gencfg) # wait for gazebo_msgs to be built
+add_dependencies(gazebo_ros_api_plugin ${catkin_EXPORTED_TARGETS})
```

Comment:
as this package generates dynamic_reconfigure files, you might want to actually write 
```add_dependencies(gazebo_ros_api_plugin ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})```
to make sure the respective header is there, i.e. `${${PROJECT_NAME}_EXPORTED_TARGETS}` ensures both former "_gencpp" and "_gencfg"

`